src/NanoProtocol.cpp

```diff
@@ -0,0 +1,30 @@
+#include "NanoProtocol.h"
+#include <stdio.h>
+#include <FileIO.h>
+#include <g3log/g3log.hpp>
+NanoProtocol::NanoProtocol(const std::string& location) {
+   if (location.find("ipc") != std::string::npos) {
+      mProtocol = Protocol::IPC;
+      mLocation = location.substr(location.find("ipc://") + 6);
+      mOwner = !FileIO::DoesFileExist(mLocation);
+   } else if (location.find("tcp") != std::string::npos){
+      mProtocol = Protocol::TCP;
+      mOwner = false;
+      mLocation = location.substr(location.find("tcp://") + 6);
+   }
+}
+NanoProtocol::~NanoProtocol() {
+   if (mProtocol == Protocol::IPC && mOwner && FileIO::DoesFileExist(mLocation)) {
+      remove(mLocation.c_str());
```

Comment:
1) this should probably be ```FileIO::RemoveAsRoot```
2) even if it doesn't have to be remove as root option please handle the result value of the operation